src/AdvisoryConsumer.cs

```diff
@@ -44,7 +45,7 @@ internal AdvisoryConsumer(Connection connection, ConsumerId consumerId) : base()
             this.info.NoLocal = true;
 
             this.connection.AddDispatcher(consumerId, this);
-            this.connection.SyncRequest(this.info);
+            this.connection.SyncRequestAsync(this.info).GetAsyncResult();
```

Comment:
From calling stack it seems this is used in async flow, but you're blocking it here. I guess this should be changed, so first, you asynchronously create a resource and then inject it here via the constructor.